aten/src/ATen/native/IndexingUtils.h

```diff
@@ -0,0 +1,135 @@
+#include <ATen/ExpandUtils.h>
+#include <ATen/native/TensorIterator.h>
+
+namespace at { namespace native {
+
+[[noreturn]]
+static void invalid_mask(const Tensor & self, int64_t idx, const Tensor & mask, int64_t maskIdx) {
+  std::stringstream ss;
+  ss << "The shape of the mask " << mask.sizes() << " at index " << maskIdx;
+  ss << " does not match the shape of the indexed tensor " << self.sizes();
+  ss << " at index " << idx;
+  AT_INDEX_ERROR(ss.str());
+}
+
+
+static std::vector<Tensor> expandTensors(const Tensor & self, TensorList indices) {
+  // Expands ByteTensor (masks) or BoolTensor (masks) into the equivalent indexing by LongTensors
+  std::vector<Tensor> result;
+  for (auto & index : indices) {
+    if (index.scalar_type() == kByte || index.scalar_type() == kBool) {
```

Comment:
cc @izdeby (I don't think there's anything wrong with this code, but I saw the instance of `kBool` and thought you'd might like to know about it)

Reply:
You will probably want to axe the `kByte` case eventually (I don't know what order you are doing these things.)